httptransport/indexer_v1.go

```diff
@@ -0,0 +1,344 @@
+			return
+		}
+
+		w.Header().Add("etag", validator)
+		defer writerError(w, &err)()
+		enc := codec.GetEncoder(w)
+		defer codec.PutEncoder(enc)
+		err = enc.Encode(report)
+	case http.MethodDelete:
+		if _, err := h.srv.DeleteManifests(ctx, d); err != nil {
+			apiError(w, http.StatusInternalServerError, "unable to delete manifest: %v", err)
+			return
+		}
+		w.WriteHeader(http.StatusNoContent)
+	}
+}
+
+func (h *IndexerV1) indexState(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodGet {
+		apiError(w, http.StatusMethodNotAllowed, "method disallowed: %s", r.Method)
+		return
+	}
+	allow := []string{"application/vnd.clair.indexstate.v1+json", "application/json"}
+	switch err := pickContentType(w, r, allow); {
+	case errors.Is(err, nil): // OK
+	case errors.Is(err, ErrMediaType):
+		apiError(w, http.StatusUnsupportedMediaType, "unable to negotiate common media type for %v", allow)
+		return
+	default:
+		apiError(w, http.StatusBadRequest, "malformed request: %v", err)
+		return
+	}
+	ctx := r.Context()
+	s, err := h.srv.State(ctx)
+	if err != nil {
+		apiError(w, http.StatusInternalServerError, "could not retrieve indexer state: %v", err)
+		return
+	}
+
+	tag := `"` + s + `"`
+	w.Header().Add("etag", tag)
+
+	if unmodified(r, tag) {
+		w.WriteHeader(http.StatusNotModified)
+		return
+	}
+
+	defer writerError(w, &err)()
+	// TODO(hank) Don't use an encoder to write out like 40 bytes of json.
+	enc := codec.GetEncoder(w)
+	defer codec.PutEncoder(enc)
+	err = enc.Encode(struct {
+		State string `json:"state"`
+	}{
+		State: s,
+	})
+}
+
+func (h *IndexerV1) affectedManifests(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		apiError(w, http.StatusMethodNotAllowed, "method disallowed: %s", r.Method)
+		return
+	}
+	allow := []string{"application/vnd.clair.affectedmanifests.v1+json", "application/json"}
+	switch err := pickContentType(w, r, allow); {
+	case errors.Is(err, nil): // OK
+	case errors.Is(err, ErrMediaType):
+		apiError(w, http.StatusUnsupportedMediaType, "unable to negotiate common media type for %v", allow)
+		return
+	default:
+		apiError(w, http.StatusBadRequest, "malformed request: %v", err)
+		return
+	}
+	ctx := r.Context()
+
+	var vulnerabilities struct {
+		V []claircore.Vulnerability `json:"vulnerabilities"`
+	}
+	dec := codec.GetDecoder(r.Body)
+	defer codec.PutDecoder(dec)
+	if err := dec.Decode(&vulnerabilities); err != nil {
+		apiError(w, http.StatusBadRequest, "failed to deserialize vulnerabilities: %v", err)
+		return
+	}
+
+	affected, err := h.srv.AffectedManifests(ctx, vulnerabilities.V)
+	if err != nil {
+		apiError(w, http.StatusInternalServerError, "could not retrieve affected manifests: %v", err)
+		return
+	}
+
+	defer writerError(w, &err)
+	enc := codec.GetEncoder(w)
+	defer codec.PutEncoder(enc)
+	err = enc.Encode(affected)
+}
+
+func init() {
+	indexerv1wrapper.init()
+}
+
+var indexerv1wrapper = &wrapper{
+	RequestCount: prometheus.NewCounterVec(
```

Comment:
We should update the grafana dashboards, I think these metric names are different,

Reply:
Yeah, I changed them to be in-line with what I could tell the best practices are.

Reply:
It looks like the 429s aren't caught with this metric as the code will return before getting into these handlers, is the idea to track them with `concurrencylimited_total`, if so I'll add a TODO to add some more charts and and SOP for excessive 429 rates

Reply:
I guess we can embed a semaphore in the handler. Whichever way you'd rather it work.

Reply:
I'm fine as it is, just making sure I understand as 429s can be used to trigger scaling up the indexers in prod environments